### PR TITLE
feat(gemma4): add dflash draft runtime

### PR DESCRIPTION
**Pull request by dusterbloom**
PR #9 of the Gemma4 split sequence. Adds Gemma4 DFlash draft graph execution, draft KV cache lifecycle, and a smoke test that exercises the full draft forward path (loader → KV cache alloc → prefill → forward). **No MTP, no γ>1 speculative decode, no server routing** — those land in PR #10/#11/#13.

Depends on #180 → #179 → #177 → #176 → #171 → #170 → #169 → #168. Stacks on \`split/gemma4-08-draft-loader-quant\`; diff includes ancestor commits until they merge.

## Scope (5 files, +761 / -8)
- \`dflash/src/gemma4_dflash_graph.cpp\` — restores the graph-builder block stripped in #180: two helpers (\`draft_swa_trunc_enabled\`, \`draft_rope\` with optional YaRN), \`build_draft_kv_prefill_graph\` (prefix-direct KV materialisation, SGLang style), \`build_gemma4_draft_graph\` (KV-cached forward returning softcapped logits).
- \`dflash/src/gemma4_target_graph.cpp\` — adds \`create_draft_kv_cache\` + \`free_draft_kv_cache\` (these live on \`GemmaTargetCache\` because the draft KV is per-target-cache). Wires \`free_draft_kv_cache(c)\` into \`free_gemma4_cache\` and \`c.draft_kv_pos = 0\` into \`reset_gemma4_cache\`.
- \`dflash/src/internal.h\` — adds the \`draft_kv_*\` fields to \`GemmaTargetCache\` (ctx, buf, K/V vectors, cap, pos) and prototypes for the four new functions.
- \`dflash/test/gemma4/smoke_gemma4_draft_forward.cpp\` — end-to-end smoke: loads target + draft, allocates draft KV, runs \`build_draft_kv_prefill_graph\`, runs \`build_gemma4_draft_graph\`, asserts non-null logits.
- \`dflash/CMakeLists.txt\` — wires the new smoke target.

## Risk: HIGH
First runtime path that actually executes the DFlash draft. Risk lives in:
- KV cache bounds: \`create_draft_kv_cache\` reserves \`sliding_window + block_size + 32\` slots; \`build_draft_kv_prefill_graph\` aborts on overflow.
- Draft RoPE: optional YaRN scaling via env (\`DFLASH_DRAFT_YARN\`), default off.
- SWA truncation: optional via env (\`DFLASH_DRAFT_SWA_TRUNC\`), default off — preserves the pre-PR#129 behavior unless opted in.

## Deferred — \`test_gemma4_dflash.cpp\` integration hunks
The full test driver in \`feature/gemma4-support\` is **2962 lines with 71 MTP references intermixed**. The \`--draft-method dflash\` flag and dflash-specific code paths cannot be cleanly carved out without (a) duplicating large chunks of the driver scaffolding into a PR-specific copy, or (b) pulling in MTP symbols this PR explicitly excludes. The full driver lands with PR #11 (MTP runtime), which already needs both \`--draft-method dflash\` and \`--draft-method mtp\` paths together.

For PR #9, the dedicated \`smoke_gemma4_draft_forward\` covers the same code paths end-to-end (loader → KV cache → prefill graph → forward graph → logits). The test driver integration is documented to maintainers and tracked.

## Forbidden-symbol audit (vs spec exclusion list)
\`\`\`
grep -cw 'mtp\|MTP'                                              → 0  (all 4 files)
grep -nE 'h_prev|MtpDrafter|MtpLayerWeights|build_mtp_step_graph'  → no matches
\`\`\`

## Review checklist (from spec)
- DFlash draft runtime works without MTP code ✓ (compiles independently — verified via syntax-only build of both .cpp files)
- Draft KV cache wrap behavior covered or deferred — **deferred**: prefix-direct prefill aborts on overflow rather than wrapping; the wrap-or-overflow policy is a follow-up decision once decode-loop logic lands. Smoke exercises happy-path only.
- Test-driver flags introduced here are only DFlash — **deferred to PR #11** per above.
- No server behavior is changed ✓.

## Validation
- \`cmake\` reconfigure picks up the new smoke target ✓
- \`g++ -fsyntax-only gemma4_dflash_graph.cpp\` via project's compile_commands flags → **exit 0**
- \`g++ -fsyntax-only gemma4_target_graph.cpp\` → **exit 0**
- Full \`cmake --build smoke_gemma4_draft_forward\` deferred to CI (no local CUDA build budget this turn)
- Runtime invocation requires both a target GGUF and a draft directory on disk.

### CODE REVIEW

**Comment by dusterbloom** [comment]
Closing: targets obsolete dflash/src/gemma4_*.cpp paths. Gemma4 moved to vendored deps/llama.cpp/src/models/gemma4-iswa.cpp; spec-decode integration is now via the common/ interface layer.
